File: src/services/resumeService.js
'use strict';

const { generatePdf } = require('../resume/generator');

class ResumeService {
  constructor(customizer) {
    this._customizer = customizer;
  }

  async buildForJob(profile, job) {
    console.log(`[Resume] Customizing for ${job.company} — ${job.title}`);
    const customized = await this._customizer.customize(
      profile, job.title, job.company, job.description
    );
    let slug = `${job.company}_${job.title}_${job.id}`.replace(/[^a-z0-9_]/gi, '_');
    slug = slug.replace(/_+/g, '_').slice(0, 60);
    const pdfPath = await generatePdf(customized, slug);
    console.log(`[Resume] PDF saved → ${pdfPath}`);
    return [customized, pdfPath];
  }
}

module.exports = { ResumeService };

File: python/src/services/resume_service.py
import re
from ..interfaces.resume_customizer import ResumeCustomizer
from ..models import UserProfile, Job, CustomizedResume
from ..resume.generator import generate_pdf


class ResumeService:
    def __init__(self, customizer: ResumeCustomizer) -> None:
        self._customizer = customizer

    async def build_for_job(self, profile: UserProfile, job: Job) -> tuple[CustomizedResume, str]:
        print(f"[Resume] Customizing for {job.company} — {job.title}")
        customized = await self._customizer.customize(
            profile, job.title, job.company, job.description
        )
        slug = re.sub(r"[^a-z0-9_]", "_", f"{job.company}_{job.title}_{job.id}", flags=re.I)
        slug = re.sub(r"_+", "_", slug)[:60]
        pdf_path = await generate_pdf(customized, slug)
        print(f"[Resume] PDF saved → {pdf_path}")
        return customized, pdf_path
